qualang_tools/optimal_weights/TimeDiffCalibrator.py

```diff
@@ -0,0 +1,153 @@
+                        'mixer': 'mixer'
+                    },
+                    'intermediate_frequency': freq,
+                    'operations': {
+                        'readout': 'readout',
+                    },
+                    "outputs": {
+                        'out1': (con_name, 1),
+                        'out2': (con_name, 2)
+                    },
+                    'time_of_flight': 32,
+                    'smearing': 0
+                },
+
+            },
+
+            "pulses": {
+
+                'readout': {
+                    'operation': 'measurement',
+                    'length': 1000,
+                    'waveforms': {
+                        'I': 'const_wf',
+                        'Q': 'zero_wf',
+
+                    },
+                    'integration_weights': {
+                        'integW1': 'integW1',
+                        'integW2': 'integW2',
+                    },
+                    'digital_marker': 'ON'
+                },
+            },
+
+            'waveforms': {
+
+                'const_wf': {
+                    'type': 'constant',
+                    'sample': 0.4
+                },
+
+                'zero_wf': {
+                    'type': 'constant',
+                    'sample': 0.0
+                },
+
+            },
+
+            'digital_waveforms': {
+                'ON': {
+                    'samples': [(1, 0)]
+                }
+            },
+
+            'integration_weights': {
+
+                'integW1': {
+                    'cosine': [1.0] * int(1000 / 4),
+                    'sine': [0.0] * int(1000 / 4)
+                },
+
+                'integW2': {
+                    'cosine': [0.0] * int(1000 / 4),
+                    'sine': [1.0] * int(1000 / 4)
+                },
+            },
+
+            'mixers': {
+
+                'mixer': [
+                    {'intermediate_frequency': freq, 'lo_frequency': 0, 'correction': [1, 0, 0, 1]}
+                ]
+            }
+        }
+
+    @staticmethod
+    def calibrate(qmm, con_name):
+        with program() as cal_phase:
+            I1 = declare(fixed)
+            Q1 = declare(fixed)
+            I2 = declare(fixed)
+            Q2 = declare(fixed)
+            I = declare(fixed)
+            Q = declare(fixed)
+
+            measure('readout', 'rr', 'adc', demod.full('integW1', I1, 'out1'),
+                    demod.full('integW2', Q1, 'out1'),
+                    demod.full('integW1', I2, 'out2'),
+                    demod.full('integW2', Q2, 'out2'))
+            assign(I, I1 + Q2)
+            assign(Q, -Q1 + I2)
+            save(I, 'I')
+            save(Q, 'Q')
+
+        freq = 1.4567e6
+        qm = qmm.open_qm(TimeDiffCalibrator._default_config(freq, con_name))
+        job = qm.simulate(cal_phase,
+                          SimulationConfig(500, simulation_interface=LoopbackInterface([('con1', 1, 'con1', 1),
```

Comment:
This is because there is some internal delay that is not known and can change between versions. This simulation extracts it...
This is actually a pretty big issue that we want to solve

Reply:
Wait, what?
The simulation is not identical to the execution?
How did you extract it from the experiment?

Reply:
Ok, just looked into this, I think that the ADC timestamps are wrong (by 20 ns), and this screws up the simulation...
In any case, I think we can just remove this file entirely and hardcode the delay according to the version. It'll also be much cleaner